widget/fit.go

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: Unlicense OR MIT
+
+package widget
+
+import (
+	"gioui.org/f32"
+	"gioui.org/layout"
+	"gioui.org/op"
+	"gioui.org/op/clip"
+)
+
+// Fit scales a widget to fit to the constraints.
+type Fit uint8
+
+const (
+	// Unscaled does not alter the scale of a widget.
+	Unscaled Fit = iota
+	// Contain scales the widget to fit exactly constraints and
+	// preserves aspect-ratio.
```

Comment:
Can we make it clearer that "fitting" means making as large as possible? Here and below.

Reply:
Tried my best.